aiutils/tftools/placeholder_management.py

```diff
@@ -34,165 +30,185 @@ class PlaceholderManager():
+            self,
+            name,
+            dtype,
+            shape=None,
+            size=None,
+            sparse=False):
         """Add placeholder.
-
-        If the sparse is True then 3 placeholders are automatically created
-        corresponding to the indices and values of the non-zero entries, and
-        shape of the sparse matrix. The user does not need to keep track of
-        these and can directly pass a sparse scipy matrix as input and a
-        Tensorflow SparseTensor object is made available for use in the graph.
-
         Args:
             name (str): Name of the placeholder.
             dtype (tf.Dtype): Data type for the placeholder.
             shape (list of ints): Shape of the placeholder.
-            sparse (bool): Specifies if the placeholder takes sparse inputs.
+            size (int) : If not None a list of placeholders is created with
+                the size being the length of the list. plh[name] returns the 
+                list of placeholders in this case.
+            sparse (bool): If True, 3 placeholders are automatically created
+                for passing in indices and values of the non-zero entries, and
+                shape of the sparse matrix. In addition it automatically adds a
+                tf.SparseTensor op which converts the 3 placeholders into a 
+                single sparse tensor. plh[name] returns a dict with 'indices',
+                'values', 'shape', and 'tensor' as keys. The first three may be
+                used to pass in inputs as part of a feed dict, and the last can 
+                be used for passing the sparse tensor to upstream ops. Instead 
+                of keeping track of 3 tensors in feed dict, a sparse matrix can 
+                be directly passed as input while using get_feed_dict method. 
+                We will refer to the dict returned by plh[name] as a sparse 
+                placeholder. 
         """
-
-        self.issparse[name] = sparse
-        if not sparse:
-            self._placeholders[name] = tf.placeholder(dtype, shape, name)
-
+        if size:
+            assert_string = 'size needs to be a positive integer'
+            assert (isinstance(size, int) and size > 0), assert_string
+            self.islist.add(name)
+            self._placeholders[name] = [None] * size
+            for i in xrange(size):
+                self._placeholders[name][i] = self.create_placeholder(
+                    name + '_' + str(i) + '_',
+                    dtype,
+                    shape=shape,
+                    sparse=sparse)
         else:
-            name_indices = name + '_indices'
-            name_values = name + '_values'
-            name_shape = name + '_shape'
+            self._placeholders[name] = self.create_placeholder(
+                name,
+                dtype,
+                shape=shape,
+                sparse=sparse)
+
+    def create_placeholder(self, name, dtype, shape=None, sparse=False):
```

Comment:
I would recommend prepending __ to this to make it private.

http://www.diveintopython.net/object_oriented_framework/private_functions.html